api/auth/auth-router.js

```diff
@@ -37,6 +52,27 @@ router.post("/login", checkUsernameExists, (req, res, next) => {
       "role_name": "admin" // the role of the authenticated user
     }
    */
+router.post("/login", checkUsernameExists, async (req, res, next) => {
+  try {
+    const token = jwt.sign(
+      {
+        subject: req.user.id,
+        username: req.user.username,
+        role_name: req.user.role_name,
+      },
+      jwtSecret,
+      {
+        expiresIn: "24h",
+      }
+    );
+    res.cookie("chocolatechipCookie", token);
+    res.status(200).json({
+      message: `${req.user.username} is back!!`,
+      token: token,
```

Comment:
daca il setezi in cookie, nu trebuie sa il transmiti ca date. Sau in cookie sau ca response, una dintre doua variante

Reply:
Ok, nu eram sigura. O sa merg pe a doua varianta, pentru ca asa se cere in conditie